sympy/parsing/c/c_parser.py

```diff
@@ -153,22 +187,13 @@ def parse_str(self, source, flags):
                 A list of sympy AST nodes
 
             """
-            file = tempfile.NamedTemporaryFile(mode = 'w+', suffix = '.cpp')
-            file.write(source)
-            file.seek(0)
             self.tu = self.index.parse(
-                file.name,
+                'test.cpp',
```

Comment:
`test.c` might be more suitable here and vice versa for `test.cpp` inside `parse_c_str`